lib/llvmopencl/LLVMFileUtils.cc

```diff
@@ -298,6 +300,9 @@ int pocl_write_file(const char *path, const char *content, uint64_t count,
     if (write(fd, content, (ssize_t)count) < (ssize_t)count)
       return errno ? -errno : -1;
 
+    if (fdatasync(fd))
```

Comment:
My man page says:
```
      On POSIX systems on which fdatasync() is available, _POSIX_SYNCHRONIZED_IO is defined in <unistd.h> to a value greater  than  0.   (See
       also sysconf(3).)
```
Can you use that macro to add a bit more portability?

Reply:
Good point, will do for next PR.